python/lsst/meas/astrom/astrometry.py

```diff
@@ -0,0 +1,257 @@
+
+        @param[in] sourceCat  catalog of sourceCat detected on the exposure (an lsst.afw.table.SourceCatalog)
+        @param[in,out] exposure  exposure whose WCS is to be fit
+            The following are read only:
+            - bbox
+            - calib (may be absent)
+            - filter (may be unset)
+            - detector (if wcs is pure tangent; may be absent)
+            The following are updated:
+            - wcs (the initial value is used as an initial guess, and is required)
+
+        This is a thin layer around \copydoc solve and returns the same data
+        """
+        bbox = exposure.getBBox()
+        exposureInfo = exposure.getInfo()
+        if not exposureInfo.hasWcs():
+            raise pipeBase.TaskError("exposure must have a WCS to use as an initial guess")
+        initWcs = exposureInfo.getWcs()
+        if not initWcs.hasDistortion() and exposureInfo.hasDetector():
+            pixelsToTanPixels = exposureInfo.getDetector().getTransformMap().get(TAN_PIXELS)
+            if pixelsToTanPixels:
+                initWcs = DistortedTanWcs(initWcs, pixelsToTanPixels)
+        calib = exposureInfo.getCalib() if exposureInfo.hasCalib() else None
+        filterName = exposureInfo.getFilter().getName() or None
+
+        retVal = self.solve(bbox=bbox, initWcs=initWcs, filterName=filterName, calib=calib)
+        exposure.setWcs(retVal.wcs)
+        return retVal
+
+    @pipeBase.timeMethod
+    def solve(self, sourceCat, bbox, initWcs, filterName=None, calib=None, exposure=None):
+        """!Fit a WCS given a source catalog and exposure metadata
+
+        @param[in] sourceCat  catalog of sourceCat detected on the exposure (an lsst.afw.table.SourceCatalog)
+        @param[in] bbox  bounding box of exposure (an lsst.afw.geom.Box2I)
+        @param[in] wcs  initial guess for WCS of exposure (an lsst.afw.image.Wcs)
+        @param[in] filterName  filter name, or None or "" if unknown (a string)
+        @param[in] calib  calibration for exposure, or None if unknown (an lsst.afw.image.Calib)
+        @param[in] exposure  exposure whose WCS is to be fit, or None; used only for the debug display
+
+        @return an lsst.pipe.base.Struct with these fields:
+        - refCat  reference object catalog of objects that overlap the exposure (with some margin)
+            (an lsst::afw::table::SimpleCatalog)
+        - matchList  list of reference object/source matchList (an lsst.afw.table.ReferenceMatchVector)
+        - initWcs  initial WCS from exposure (possibly tweaked) (an lsst.afw.image.Wcs)
+        - wcs  fit WCS (an lsst.afw.image.Wcs)
+        """
+        import lsstDebug
+        debug = lsstDebug.Info(__name__)
+
+        loadRes = self.refObjLoader.loadObjectsInBBox(
+            bbox = bbox,
+            wcs = initWcs,
+            filterName = filterName,
+            calib = calib,
+        )
+        if debug.display:
+            frame = debug.frame if isinstance(debug.frame, int) else 1
```

Comment:
I don't particularly like the `isinstance` check here (especially since Python 2 also has `long`, which isn't a subclass of `int`); what could `debug.frame` be other than an integer or `None`?